src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1SpaceSearchTest.kt

```diff
@@ -103,4 +347,217 @@ class Cas1SpaceSearchTest : InitialiseDatabasePerClassTestBase() {
     assertThat(actual.premises!!.totalSpaceCount).isEqualTo(expected.rooms.flatMap { it.beds }.count())
     assertThat(actual.premises!!.premisesCharacteristics).isEmpty()
   }
+
+  @ParameterizedTest
+  @EnumSource
+  fun `Filtering APs by characteristic only returns APs with that characteristic`(characteristic: Cas1SpaceCharacteristic) {
+    postCodeDistrictFactory.produceAndPersist {
+      withOutcode("SE1")
+      withLatitude(-0.07)
+      withLongitude(51.48)
+    }
+
+    `Given a User` { _, jwt ->
+      val expectedPremises = approvedPremisesEntityFactory.produceAndPersistMultipleIndexed(5) {
+        withYieldedProbationRegion {
+          probationRegionEntityFactory.produceAndPersist {
+            withYieldedApArea {
+              apAreaEntityFactory.produceAndPersist()
+            }
+          }
+        }
+        withYieldedLocalAuthorityArea {
+          localAuthorityEntityFactory.produceAndPersist()
+        }
+        withLatitude((it * -0.01) - 0.08)
+        withLongitude((it * 0.01) + 51.49)
+        withCharacteristicsList(listOf(characteristic.asCharacteristicEntity()))
+      }
+
+      expectedPremises.forEach {
+        roomEntityFactory.produceAndPersist {
+          withPremises(it)
+          withCharacteristicsList(listOf(characteristic.asCharacteristicEntity()))
+        }
+      }
+
+      println("===\nPremises:\n${expectedPremises.map { it.id to it.characteristics }}\n===")
```

Comment:
if still required could this please be a log output instead?

Reply:
This has been removed as it's no longer necessary.